src/fleet/azext_fleet/_helpers.py

```diff
@@ -148,3 +152,15 @@ def _load_kubernetes_configuration(filename):
         raise
     except (yaml.parser.ParserError, UnicodeDecodeError) as ex:
         raise CLIError(f'Error parsing {filename} ({str(ex)})') from ex
+
+
+def validate_subnet(cmd, subnet_id):
```

Comment:
consider putting this in _validators.py ?

Reply:
maybe add a comment about why we check the provider -- it's to catch the obvious other case that would cause the SP not to exist and `add_role_assignment` to hence fail